Remove tycho.mode=maven

We should no longer support `tycho.mode=maven` it was often used to package an update-site in a regular maven build or to build "mixed reactors", for both we already have better solutions now.

It indeed worked to add the tp-module in the `tycho.mode=maven` build. To not interfere with the regular build I added it in the profile that is activated only when `tycho.mode=maven` was used, please see the second commit in https://github.com/eclipse-m2e/m2e-core/pull/589/
This has the positive effect that only the profile has to be activated and the additional `tycho.mode=maven` property can be skipped, which makes the build command simpler. I think this is an 'workaround' for the better, since I expect a profile to be used in cases similar to the one of M2E anyway.

The only problem left is, that it does not work when the `MANIFEST.MF` was not yet generated and fails with the exception below.

Do you think the project setup could fall back to the pom.xml (which I think has to be present in such cases)?

```
[INFO] Scanning for projects...
[ERROR] Internal error: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST C:\dev\git\eclipse.m2e-core\m2e-maven-runtime\org.eclipse.m2e.archetype.common\META-INF\MANIFEST.MF: Manifest file not found -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST C:\dev\git\eclipse.m2e-core\m2e-maven-runtime\org.eclipse.m2e.archetype.common\META-INF\MANIFEST.MF: Manifest file not found
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:120)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:972)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
Caused by: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST C:\dev\git\eclipse.m2e-core\m2e-maven-runtime\org.eclipse.m2e.archetype.common\META-INF\MANIFEST.MF: Manifest file not found
	at org.eclipse.tycho.core.osgitools.DefaultBundleReader.loadManifestFromDirectory(DefaultBundleReader.java:93)
	at org.eclipse.tycho.core.osgitools.DefaultBundleReader.doLoadManifest(DefaultBundleReader.java:60)
	at org.eclipse.tycho.core.osgitools.DefaultBundleReader.loadManifest(DefaultBundleReader.java:51)
	at org.eclipse.tycho.core.osgitools.OsgiBundleProject.readArtifactKey(OsgiBundleProject.java:194)
	at org.eclipse.tycho.core.osgitools.OsgiBundleProject.setupProject(OsgiBundleProject.java:176)
	at org.eclipse.tycho.core.resolver.DefaultTychoResolver.setupProject(DefaultTychoResolver.java:89)
	at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead(TychoMavenLifecycleParticipant.java:111)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:264)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	... 11 more
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/InternalErrorException
```

--- a/RELEASE_NOTES.md
+++ b/RELEASE_NOTES.md
@@ -59,6 +59,9 @@ Please note that we use another new feature from Maven 3.8.5 here, where you can
 
 You can now control your Tycho version for `.mvn/extensions.xml` and your `pom.xml` in one place and still override it on the commandline with `-Dtycho-version=...`
 
+### Support for -Dtycho.mode=maven removed
+
+The property has never been properly documented and excluding tycho resolution is risky if one is not 100% sure what the effects are (e.g  using it with maven-dependency-plugin will result will many dependency not being listed at all). Clean phase is manually excluded by Tycho as we clearly know that it doesn't need resolution, if there is the need for more such exclusions they should be done similarly.
 
 ### Support for non-modular JVMs no longer top tier
 

--- a/tycho-bundles/tycho-bundles-external/pom.xml
+++ b/tycho-bundles/tycho-bundles-external/pom.xml
@@ -31,7 +31,7 @@
 		1. Run "mvn clean install" in tycho-bundles folder. This is necessary to populate local repo with information
 		   about org.eclipse.tycho.noopsecurity bundle
 		2. If necessary, change p2 repository url in tycho-bundles/pom.xml
-		3. In this folder, run mvn -Dtycho.mode=maven -Dtycho-version=0.11.0-SNAPSHOT org.sonatype.tycho.extras:tycho-version-bump-plugin:0.11.0-SNAPSHOT:update-product
+		3. In this folder, run mvn -Dtycho-version=0.11.0-SNAPSHOT org.sonatype.tycho.extras:tycho-version-bump-plugin:0.11.0-SNAPSHOT:update-product
 		4. Edit tycho-p2-runtime.product and remove version lock on org.eclipse.tycho.noopsecurity bundle
 
 		Luckily, we don't need to include new p2 version too often.

--- a/tycho-core/src/main/java/org/eclipse/tycho/core/maven/TychoMavenLifecycleParticipant.java
+++ b/tycho-core/src/main/java/org/eclipse/tycho/core/maven/TychoMavenLifecycleParticipant.java
@@ -15,7 +15,6 @@ package org.eclipse.tycho.core.maven;
 
 import java.io.File;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
@@ -57,10 +56,9 @@ import org.eclipse.tycho.resolver.TychoResolver;
 public class TychoMavenLifecycleParticipant extends AbstractMavenLifecycleParticipant {
 
     private static final String TYCHO_GROUPID = "org.eclipse.tycho";
-    private static final Set<String> TYCHO_PLUGIN_IDS = new HashSet<>(Arrays.asList("tycho-maven-plugin",
-            "tycho-p2-director-plugin", "tycho-p2-plugin", "tycho-p2-publisher-plugin", "tycho-p2-repository-plugin",
-            "tycho-packaging-plugin", "tycho-source-plugin", "tycho-surefire-plugin",
-            "tycho-versions-plugin", "tycho-compiler-plugin"));
+    private static final Set<String> TYCHO_PLUGIN_IDS = Set.of("tycho-maven-plugin", "tycho-p2-director-plugin",
+            "tycho-p2-plugin", "tycho-p2-publisher-plugin", "tycho-p2-repository-plugin", "tycho-packaging-plugin",
+            "tycho-source-plugin", "tycho-surefire-plugin", "tycho-versions-plugin", "tycho-compiler-plugin");
     private static final String P2_USER_AGENT_KEY = "p2.userAgent";
     private static final String P2_USER_AGENT_VALUE = "tycho/";
 
@@ -268,8 +266,7 @@ public class TychoMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 
     private boolean disableLifecycleParticipation(MavenSession session) {
         // command line property to disable Tycho lifecycle participant
-        return "maven".equals(session.getUserProperties().get("tycho.mode"))
-                || session.getUserProperties().containsKey("m2e.version")
+        return session.getUserProperties().containsKey("m2e.version")
                 // disable for 'clean-only' builds. Consider that Maven can be invoked without explicit goals, if default goals are specified
                 || (!session.getGoals().isEmpty() && CLEAN_PHASES.containsAll(session.getGoals()));
     }

